File: examples/java/kora-java-grpc-client/src/main/java/io/koraframework/example/grpc/client/RootService.java
package io.koraframework.example.grpc.client;

import io.koraframework.common.annotation.Component;
import io.koraframework.common.annotation.Root;
import io.koraframework.generated.grpc.UserServiceGrpc;

@Root
@Component
public final class RootService {

    private final UserServiceGrpc.UserServiceBlockingStub userService;

    public RootService(UserServiceGrpc.UserServiceBlockingStub userService) {
        this.userService = userService;
    }

    public UserServiceGrpc.UserServiceBlockingStub service() {
        return userService;
    }
}

File: examples/kotlin/kora-kotlin-grpc-client/src/main/kotlin/io/koraframework/kotlin/example/grpc/client/RootService.kt
package io.koraframework.kotlin.example.grpc.client

import io.grpc.*
import org.slf4j.LoggerFactory
import io.koraframework.common.annotation.Component
import io.koraframework.common.annotation.Tag
import io.koraframework.common.annotation.Root
import io.koraframework.generated.grpc.UserServiceGrpc

@Root
@Component
class RootService(private val userService: UserServiceGrpc.UserServiceBlockingStub) {
    fun service(): UserServiceGrpc.UserServiceBlockingStub = userService
}

